src/transforms/scaling.jl

```diff
@@ -0,0 +1,58 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Scaling(sx, sy, sz)
```

Comment:
Please use a vector of scaling factors instead to avoid a 3D-specific implementation.

Reply:
Do you mean a vector for the definition of the `struct`, and we keep the constructor with the three scaling factors? I prefer the three scaling factors because the vector made of the scaling factors has no geometrical interpretation, this is just a list of numbers, not a geometric object.

Reply:
Yes, the struct itself stores a vector of scaling parameters in a way that is dimension-agnostic. Then we can have as many outer constructors as needed for convenience that take list of arguments and convert to the internal vector.